test/parser.specs.js

```diff
@@ -291,6 +317,14 @@ describe('odata.parser grammar', function () {
         assert.equal(ast.error, 'invalid $format parameter');
     });
 
+    it('should parse $search okay', function () {
+        var ast = parser.parse('$search="foo bar baz"');
```

Comment:
does it handle `"` being in the input (i.e. escape it...or should be handle that in the mapper?)